src/Adapters/NLogTarget.Shared/ApplicationInsightsTarget.cs

```diff
@@ -50,6 +51,16 @@ internal TelemetryClient TelemetryClient
         }
 
         /// <summary>
+        /// Flush any pending log messages asynchronously (in case of asynchronous targets).
+        /// </summary>
+        /// <param name="asyncContinuation">The asynchronous continuation.</param>
+        protected override void FlushAsync(AsyncContinuation asyncContinuation)
+        {
+            this.telemetryClient.Flush();
```

Comment:
is it expected to block the thread by `telemetryClient.Flush()` inside the `FlushAsync` call? Maybe it should be wrapped into the `new Task(() => {tc.Flush()} ).Start()` and hook it up into the asyncContinuation?